### PR TITLE
Drawer Nav Item using Link as the component

### LINKED ISSUE #182: Drawer should take a component prop to render the menu items in
#### Describe the desired behavior

`<Drawer />` should be able to take a `component` prop (probably as those inheritable props) that can render its drawer item as other components

By default, use `Link` from `react-router-dom` to pass in to the info list item

#### Describe the current behavior

It is using the default ListItem behavior, which renders our list items as either a `<li>` or a `<div>`. It does not allow us to right-click and choose "open in a new tab". 

#### Is this request related to a current issue?

stem from discussions here: https://github.com/pxblue/react-component-library/issues/113

### PR DESCRIPTION
**Pull request by huayunh**
Fixes #182

#### Changes proposed in this Pull Request:

I am not 100% sure if this change to the drawer component is necessary. I just realized that you are able to do `{ to: '/overview', component: Link }` to `InfoListItemProps` and get it to work. Doing what I am doing here might make it a breaking change. 

- Make drawer render a [`<Link>`](https://reactrouter.com/web/api/Link) component from `react-router-dom` when a `to` attribute in the `InfoListItemProps` is present
- `react-router-dom` is now listed as a peer dependency
- Update the ILI & drawer docs to offer suggestions on how to render links

#### Screenshots:

To verify my work, if I do this to line 107 of `./demos/showcase/src/router/NavigationDrawer.tsx`, I will get this screenshot
```tsx
                        {
                            title: titleList[0],
                            itemID: titleList[0],
                            icon: <ListIcon className={clsx({ [classes.iconFlip]: rtl })} />,
                            onClick: (): void => {
                                // navigate(0);
                                if (xsDown) dispatch({ type: CLOSE_DRAWER });
                            },
                            // @ts-ignore
                            InfoListItemProps: { to: '/overview' },
                        },
```
image

it will be able to update the current location, can be opened in a new tab with the correct address, the tooltip thing shows up in the bottom left corner of the browser, and upon clicking the browser does not refresh (which suggest that we are using `<Link>` here). Note that the background shading is controlled by a different mechanism and is not updated.

As a comparison, other drawer nav items stay the same way: 
image

### CODE REVIEW

**Comment by huayunh** [comment]
I am adding the `component` prop to InfoListItem instead because that's where the problem stems... 

Although it turns out you will still have to ts-ignore something, because `to` is not part of the InfoListItemProps.

File: docs/InfoListItem.md [hunk 1]
```diff
@@ -70,3 +70,29 @@ You can override the classes used by PX Blue by passing a `classes` prop. It sup
 | statusStripe   | Styles applied to the status stripe element         |
 | subtitle       | Styles applied to the subtitle element              |
 | title          | Styles applied to the title element                 |
+
+## Tips
+
+You can render the `<InfoListItem>` as a link by using the [`<Link>`](https://reactrouter.com/web/api/Link) component from `react-router-dom` (recommended), or the native anchor tag `<a>`.
```
Comment (joebochill) [inline docs/InfoListItem.md:76]:
```suggestion
You can render the `<InfoListItem>` as a link by using the [`<Link>`](https://reactrouter.com/web/api/Link) component from `react-router-dom` (recommended), or the native anchor tag `<a>`. This allows you to perform helpful actions such as opening a link in a new browser tab.
```

**Comment by huayunh** [comment]
Update: 

From our discussion, we came to realize that MUI is changing their ListItemProps dynamically with some generic type tricks. Here are two screenshots from our experiment:

MicrosoftTeams-image (2)
MicrosoftTeams-image (3)

We agreed to close this PR for now. I will add a little more explanation in the enhancement.

**Comment by huayunh** [comment]
@joebochill should be good to go

File: docs/Drawer.md [hunk 2]
```diff
@@ -345,6 +345,38 @@ The following props control the NavGroup and thus only apply to `<Drawer>`, and
...
+
+<Drawer ... >
+    <DrawerBody>
+        <DrawerNavGroup
+            item={[
+                // rendered as <Link>
+                {
+                    title: 'Overview',
+                    itemID: '0',
+                    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (joebochill) [reply docs/Drawer.md:364]: Let's log a separate issue to dig deeper into this

**Review by joebochill — APPROVED** [review]

**Pull request merged**